data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/NodeSyncingHandlerTest.java

```diff
@@ -0,0 +1,69 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.beaconrestapi.beaconhandlers;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.primitives.UnsignedLong;
+import io.javalin.http.Context;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.artemis.beaconrestapi.schema.SyncingResponse;
+import tech.pegasys.artemis.provider.JsonProvider;
+import tech.pegasys.artemis.sync.SyncService;
+import tech.pegasys.artemis.sync.SyncStatus;
+import tech.pegasys.artemis.sync.SyncingStatus;
+
+public class NodeSyncingHandlerTest {
+  private Context context = mock(Context.class);
+  private final JsonProvider jsonProvider = new JsonProvider();
+  private final SyncService syncService = mock(SyncService.class);
+
+  @Test
+  public void shouldReturnTrueWhenSyncing() throws Exception {
+    final boolean isSyncing = true;
+    final UnsignedLong startSlot = UnsignedLong.ONE;
+    final UnsignedLong currentSlot = UnsignedLong.valueOf(5);
+    final UnsignedLong highestSlot = UnsignedLong.valueOf(10);
+    SyncingStatus syncingStatus =
+        new SyncingStatus(isSyncing, new SyncStatus(startSlot, currentSlot, highestSlot));
+    tech.pegasys.artemis.beaconrestapi.schema.SyncingStatus syncStatus =
+        new tech.pegasys.artemis.beaconrestapi.schema.SyncingStatus(
+            startSlot, currentSlot, highestSlot);
+    SyncingResponse syncingResponse = new SyncingResponse(isSyncing, syncStatus);
+    when(syncService.getSyncStatus()).thenReturn(syncingStatus);
+    NodeSyncingHandler handler = new NodeSyncingHandler(syncService, jsonProvider);
+    handler.handle(context);
+    verify(context).result(jsonProvider.objectToJSON(syncingResponse));
+  }
+
+  @Test
+  public void shouldReturnFalseWhenNotSyncing() throws Exception {
+    final boolean isSyncing = true;
+    final UnsignedLong startSlot = UnsignedLong.ZERO;
+    final UnsignedLong currentSlot = UnsignedLong.ZERO;
+    final UnsignedLong highestSlot = UnsignedLong.ZERO;
+    SyncingStatus syncingStatus =
+        new SyncingStatus(isSyncing, new SyncStatus(startSlot, currentSlot, highestSlot));
+    tech.pegasys.artemis.beaconrestapi.schema.SyncingStatus syncStatus =
+        new tech.pegasys.artemis.beaconrestapi.schema.SyncingStatus(
+            startSlot, currentSlot, highestSlot);
+    SyncingResponse syncingResponse = new SyncingResponse(isSyncing, syncStatus);
+    when(syncService.getSyncStatus()).thenReturn(syncingStatus);
+    NodeSyncingHandler handler = new NodeSyncingHandler(syncService, jsonProvider);
+    handler.handle(context);
+    verify(context).result(jsonProvider.objectToJSON(syncingResponse));
```

Comment:
theres a lot of setup in this unit test. i'd probably at least suggest moving the 'when' down next to the handle, and putting in a line break - then the 3 lines down the bottom are the test, the rest above is the setup.
Potentially if we rename the SyncingStatus that's a response (SyncingStatusResponse), it might reduce some of the noise too.

Reply:
simplified